src/types/types.ts

```diff
@@ -2591,6 +2591,93 @@ export interface DebugPrestate {
   storage: Record<string, string>;
 }
 
+/*
+ * SDK Representation of a user operation and bundler types
+ */
+export interface UserOperation {
+  sender: string;
+  nonce: string;
+  initCode: string;
+  callData: string;
+  callGasLimit: number;
+  verificationGasLimit: number;
+  preVerificationGas: number;
+  maxFeePerGas: number;
+  maxPriorityFeePerGas: number;
+  paymasterAndData: string;
+  signature: string;
+  entryPoint: string;
+  blockNumber: number;
+  blockHash: string;
+  transactionHash: string;
+}
+
+export interface UserOperationRequest {
+  sender: string;
+  nonce: string;
+  initCode: string;
+  callData: string;
+  callGasLimit: number;
+  verificationGasLimit: number;
+  preVerificationGas: number;
+  maxFeePerGas: number;
```

Comment:
wanted to double check, are all fields here required? Aren't fields like `maxFeePerGas` and `maxPriorityFeePerGas` usually optional?

Reply:
Hmm not according to our api: https://github.com/OMGWINNING/alchemy-api-executors/blob/3b6054e7/evm-service/src/main/java/alchemy/evmservice/methods/rundler/EthSendUserOperation.java#L28

Reply:
Let me ask @dancoombs and @dphilipson

Reply:
Looks like we do https://alchemyinsights.slack.com/archives/C04RHSDSP9U/p1683141735130749

Reply:
Does this type represent input that the user provides, or the actual data sent over the wire to the bundler? I'm assuming the former, since otherwise all the numbers should be hex strings. In that case, from the linked thread I would consider making all the gas fields optional, and if not provided have the sdk fill them in.